qiskit/optimization/converters/quadratic_program_to_ising.py

```diff
@@ -114,6 +114,8 @@ def encode(self, op: QuadraticProgram) -> Tuple[WeightedPauliOperator, float]:
             shift += weight
 
         # Remove paulis whose coefficients are zeros.
-        qubit_op = WeightedPauliOperator(paulis=pauli_list)
+        qubit_op = sum(PauliOp(pauli, coeff=coeff) for coeff, pauli in pauli_list)
+        if isinstance(qubit_op, SummedOp):
```

Comment:
Is there are reason why the isinstance needs to be here? If the expected outcome is an opflow operator then reduce is part of the base class interface - though given the docstring in the base class I do not know why it would not just return self for an implementation there (at present its abstract and raises a not implemented error)

Reply:
What you said last is the reason -- if sum() returns a PauliOp, which could happen if the pauli_list has only one entry, then reduce is not implemented. We could also add it as trivial method on the OperatorBase, do you prefer this solution?

Reply:
reduce is implemented for PauliOp though - reduce is implemented in PrimitiveOp as is described in BaseOperator docstring. The BaseOperator just does not seem to implement it as it describes itself, whether this was done as part of dev to ensure other better reduce implementations were done on sub-classes I am unsure. Either way it seems that the base class could implement what it states after all a default of returning itself seems fine.

Reply:
I remember the reason: the `sum()` above could return 0 if the `pauli_list` is empty. I changed for a test of `isinstance(qubit_op, OperatorBase)`.

Reply:
If the pauli list is empty and as you have it qubit_op becomes the integer 0 shouldn't the case of the empty pauli list be detected and an empty operator for qubit_op be created for the return value instead of letting an integer (of value 0) get returned. The method states it returns an OperatorBase so it should do that even if its an empty operator.

Reply:
Yes, agreed. Though I'm not sure an empty operator is possible, since `OperatorBase` is abstract and all other operator types require an input (the constructor fails upon the default argument `None` -- which we should change). So we could instead just return an identity Pauli operator of correct size.